src/otx/algo/instance_segmentation/losses/maskdino_criterion.py

```diff
@@ -0,0 +1,425 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+"""MaskFormer criterion."""
+from __future__ import annotations
+
+import torch
+import torch.distributed
+import torch.nn.functional as f
+from torch import Tensor, nn
+
+from otx.algo.instance_segmentation.losses import HungarianMatcher
+from otx.algo.instance_segmentation.utils import box_ops
+from otx.algo.instance_segmentation.utils.utils import get_uncertain_point_coords_with_randomness, point_sample
+
+
+def sigmoid_focal_loss(
+    inputs: Tensor,
+    targets: Tensor,
+    num_boxes: float,
+    alpha: float = 0.25,
+    gamma: float = 2,
+) -> Tensor:
+    """Loss used in RetinaNet for dense detection: https://arxiv.org/abs/1708.02002.
+
+    Args:
+        inputs: A float tensor of arbitrary shape.
+                The predictions for each example.
+        targets: A float tensor with the same shape as inputs. Stores the binary
+                 classification label for each element in inputs
+                (0 for the negative class and 1 for the positive class).
+        alpha: (optional) Weighting factor in range (0,1) to balance
+                positive vs negative examples. Default = -1 (no weighting).
+        gamma: Exponent of the modulating factor (1 - p_t) to
+               balance easy vs hard examples.
+
+    Returns:
+        Loss tensor
+    """
+    prob = inputs.sigmoid()
+    ce_loss = f.binary_cross_entropy_with_logits(inputs, targets, reduction="none")
+    p_t = prob * targets + (1 - prob) * (1 - targets)
+    loss = ce_loss * ((1 - p_t) ** gamma)
+
+    if alpha >= 0:
+        alpha_t = alpha * targets + (1 - alpha) * (1 - targets)
+        loss = alpha_t * loss
+
+    return loss.mean(1).sum() / num_boxes
+
+
+def dice_loss(
```

Comment:
It looks same with below
https://github.com/openvinotoolkit/training_extensions/blob/7773abb5bf6c44e9d180a91b861fc895ecd6e6d2/src/otx/algo/instance_segmentation/losses/dice_loss.py#L17-L74

Reply:
I will remove the duplication in the next phase.